src/bounded_vec.rs

```diff
@@ -94,49 +174,100 @@ impl<T, const L: usize, const U: usize> BoundedVec<T, L, U> {
     /// assert_eq!(data.to_vec(), vec![1u8,2]);
     /// ```
     pub fn to_vec(self) -> Vec<T> {
-        self.into()
+        self.inner
     }
 
-    /// Returns the number of elements in the vector
+    /// Extracts a slice containing the entire vector.
     ///
     /// # Example
     /// ```
     /// use bounded_vec::BoundedVec;
     /// use std::convert::TryInto;
     ///
-    /// let data: BoundedVec<u8, 2, 4> = vec![1u8,2].try_into().unwrap();
-    /// assert_eq!(data.len(), 2);
+    /// let data: BoundedVec<_, 2, 8> = vec![1u8, 2].try_into().unwrap();
+    /// assert_eq!(data.as_slice(), &[1u8,2]);
     /// ```
-    pub fn len(&self) -> usize {
-        self.inner.len()
+    pub fn as_slice(&self) -> &[T] {
+        self.inner.as_slice()
     }
 
-    /// Always returns `false` (cannot be empty)
+    /// Returns a reference for an element at index or `None` if out of bounds
     ///
     /// # Example
+    ///
     /// ```
     /// use bounded_vec::BoundedVec;
-    /// use std::convert::TryInto;
+    /// let data: BoundedVec<u8, 2, 8> = [1u8,2].into();
+    /// let elem = *data.get(1).unwrap();
+    /// assert_eq!(elem, 2);
+    /// ```
+    pub fn get(&self, index: usize) -> Option<&T> {
+        self.inner.get(index)
+    }
+
+    /// Returns an iterator
+    pub fn iter(&self) -> Iter<T> {
+        self.inner.iter()
+    }
+
+    /// Returns an iterator that allows to modify each value
+    pub fn iter_mut(&mut self) -> IterMut<T> {
+        self.inner.iter_mut()
+    }
+}
+
+impl<T, const L: usize, const U: usize> BoundedVec<T, L, U, witnesses::NonEmpty<L, U>> {
+    /// Creates new BoundedVec or returns error if items count is out of bounds
     ///
-    /// let data: BoundedVec<_, 2, 8> = vec![1u8, 2].try_into().unwrap();
-    /// assert_eq!(data.is_empty(), false);
+    /// # Parameters
+    ///
+    /// * `items` - vector of items within bounds
+    ///
+    /// # Errors
+    ///
+    /// * `LowerBoundError` - if `items`` len is less than L (lower bound)
+    /// * `UpperBoundError` - if `items`` len is more than U (upper bound)
+    ///
+    /// # Example
     /// ```
-    pub fn is_empty(&self) -> bool {
-        false
+    /// use bounded_vec::BoundedVec;
+    /// use bounded_vec::witnesses;
+    /// let data: BoundedVec<_, 2, 8, witnesses::NonEmpty<2, 8>> =
+    ///     BoundedVec::<_, 2, 8, witnesses::NonEmpty<2, 8>>::from_vec(vec![1u8, 2]).unwrap();
+    /// ```
+    pub fn from_vec(items: Vec<T>) -> Result<Self, BoundedVecOutOfBounds> {
+        let _witness = witnesses::non_empty::<L, U>();
```

Comment:
Based on my testing these calls need to be in a const { } block, otherwise the panic still occurs at runtime

Reply:
added.

Reply:
![image](https://github.com/user-attachments/assets/66a3fe7a-3225-4bfa-9e9b-7bd91a0c6063)

Reply:
there is no force it to be compile time only....

Reply:
thinking on hack

Reply:
just moved const inside.

`cargo build --tests` fails well